### PR TITLE
Nazar zvarych branch

### PR DESCRIPTION
**Pull request by Bandura2**
Home work:
Lab 5

### CODE REVIEW

File: Laba_5.h [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    os <<"Size: "<< object.size <<endl;
+    os <<"Color: "<< object.color <<endl;
+    os <<"Price: "<< object.price_on_hryvna <<endl;
+    os <<"Quantity in stock: "<< object.quantity_in_stock <<endl;
+    os <<"Material: "<< object.material <<endl;
+    os <<"Quantity sales: "<< object.quantity_sales <<endl << endl;
+    return os;
+}
+
+float comparing(float array_prices[], vector<Sneaker> comparable_vector,int size)
```
Comment (artur-zhdan) [inline Laba_5.h]: Function name should start with uppercase

File: Laba_5.h [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+    ~Sneaker () {}
+};
+
+class SportShoesStore
+{
+public:
+    SportShoesStore () {}
+
+    void Sort(float arr[], int size)
```
Comment (artur-zhdan) [inline Laba_5.h]: unclear function functionality - what is the rule for sorting? Please, provide a brief comment about it

File: Laba_5.h [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        {
+            for(int second_temporary_index = 0; second_temporary_index < size; second_temporary_index++)
+            {
+                if(array_prices[temporary_index] == comparable_vector[second_temporary_index].price_on_hryvna)
+                    cout << comparable_vector[second_temporary_index];
+
+            }
+        }
+    }
+int comparing(int array_sales[], vector<Sneaker> comparable_vector,int size)
```
Comment (artur-zhdan) [inline Laba_5.h]: Function name should start with uppercase.
In addition you may prefer using templates for function declaration if you need it for different types.

File: main.cpp [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+int main()
+{
+    vector<Sneaker> sneakers;
+    cout <<"Enter number of objects:"<<endl;
+    int num_object;
+    cin >> num_object;
+    cout <<"----------------------------------------------------------"<< endl;
+
+  for(int temporary_index = 0; temporary_index < num_object; temporary_index++)
```
Comment (artur-zhdan) [inline main.cpp]: add four spaces/tab

File: main.cpp [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    {
+        sneakers.push_back(Sneaker());
+    }
+
+    for(int temporary_index = 0; temporary_index < num_object; temporary_index++)
```
Comment (artur-zhdan) [inline main.cpp]: better use just "i" as index

File: main.cpp (hunk 4 above)
Comment (artur-zhdan) [inline main.cpp]: use "i" as index

File: main.cpp [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    {
+        cout << "\nSneaker "<< temporary_index + 1 <<" :"<<endl;
+        cin >> sneakers[temporary_index];
+    }
+    cout <<"----------------------------------------------------------"<< endl;
+
+    float *array_prices = new float[num_object];
+    for(int temporary_index = 0; temporary_index < num_object; temporary_index++)
```
Comment (artur-zhdan) [inline main.cpp]: "i"

File: main.cpp [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    {
+        array_prices[temporary_index] = sneakers[temporary_index].price_on_hryvna;
+    }
+
+    SportShoesStore object_for_sort;
```
Comment (artur-zhdan) [inline main.cpp]: if you use this object just for two methods, without fields, you should declare functions separately. no need for a redundant object

File: main.cpp [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    cout<<"If you want to see the top, enter 1, else program will be closed"<<endl;
+    int choise = 0;
+    cin >> choise;
+
+    if (choise == 1)
+    {
+        cout <<"----------------------------------------------------------"<< endl;
+        cout <<"Top sales :\n"<<endl;
+        int *array_sales = new int[num_object];
+        for(int temporary_index = 0; temporary_index < num_object; temporary_index++)
```
Comment (artur-zhdan) [inline main.cpp]: "i"

File: main.cpp [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+    object_for_sort.Sort(array_prices, num_object);
+
+    cout<<"Catalog:\n" << endl;
+
+    comparing(array_prices, sneakers, num_object);
+
+    cout<<"The catalog has been sorted by price"<<endl;
+    cout <<"----------------------------------------------------------"<< endl;
+    cout<<"If you want to see the top, enter 1, else program will be closed"<<endl;
+    int choise = 0;
```
Comment (artur-zhdan) [inline main.cpp]: choice not choise

**Review by artur-zhdan — CHANGES_REQUESTED** [review]
Overall the code is good. just do some minor changes

**Review by artur-zhdan — APPROVED** [review]

File: Laba_5.h [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+
+    ~Sneaker () {}
+};
+
+class SportShoesStore
+{
+public:
+    SportShoesStore () {}
+
+    void Sort_array(float arr[], int size)  /*Bubblesort*/
```
Comment (zveres-ss) [inline Laba_5.h:29]: either SortArray or sort_array

**Review by zveres-ss — CHANGES_REQUESTED** [review]

**Pull request merged**